internal/pkg/stream/ecs.go

```diff
@@ -32,6 +34,15 @@ type ECSDeployment struct {
 	FailedCount     int
 	PendingCount    int
 	RolloutState    string
+	CreatedAt       time.Time
+}
+
+func (d ECSDeployment) isPrimary() bool {
+	return d.Status == ecsPrimaryDeploymentStatus
+}
+
+func (d ECSDeployment) done() bool {
+	return d.DesiredCount == d.RunningCount && d.RolloutState != rollOutInProgress
```

Comment:
Should this be `d.RolloutState == "COMPLETED"` in case it failed?

Reply:
I think we can consider a particular deployment done when it's not in progress (either failed or completed). Because if it fails, then the previously active deployment becomes the primary and becomes completed.